Spigot-API-Patches/0221-Create-HoverEvent-from-ItemStack-Entity.patch

```diff
@@ -0,0 +1,108 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: ysl3000 <yannicklamprecht@live.de>
+Date: Mon, 6 Jul 2020 22:17:37 +0200
+Subject: [PATCH] Create-HoverEvent-from-ItemStack-Entity
+
+
+diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
+index bfe842364ee0a4bf39dacdbb6972477d57a4ef8a..7998d279cc0e1d9fe64ba2c600a47b8995f2376f 100644
+--- a/src/main/java/org/bukkit/Bukkit.java
++++ b/src/main/java/org/bukkit/Bukkit.java
+@@ -13,6 +13,7 @@ import java.util.Set;
+ import java.util.UUID;
+ import java.util.function.Consumer;
+ import java.util.logging.Logger;
++import net.md_5.bungee.api.chat.BaseComponent; // Paper
+ import org.bukkit.Warning.WarningState;
+ import org.bukkit.advancement.Advancement;
+ import org.bukkit.block.data.BlockData;
+@@ -1781,6 +1782,41 @@ public final class Bukkit {
+     public static com.destroystokyo.paper.entity.ai.MobGoals getMobGoals() {
+         return server.getMobGoals();
+     }
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that ItemStack for displaying.
++     *
++     * @param itemStack ItemStack to create the HoverEvent for
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that ItemStack
++     */
++    @NotNull
++    public static net.md_5.bungee.api.chat.hover.content.Content hoverContentOf(@NotNull ItemStack itemStack) {
++        return server.hoverContentOf(itemStack);
++    }
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity} for displaying.
++     *
++     * @param entity     Entity to create the HoverEvent for
++     * @param customName a custom name that should be displayed, if not passed entity name will be displayed
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity}
++     */
++    @NotNull
++    public static net.md_5.bungee.api.chat.hover.content.Content hoverContentOf(@NotNull Entity entity, @Nullable String customName) {
++        return server.hoverContentOf(entity, customName);
++    }
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity} for displaying.
++     *
++     * @param entity     Entity to create the HoverEvent for
++     * @param customName a custom name that should be displayed, if not passed entity name will be displayed
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity}
++     */
++    @NotNull
++    public static net.md_5.bungee.api.chat.hover.content.Content hoverContentOf(@NotNull Entity entity, @Nullable BaseComponent customName) {
```

Comment:
Should this not be `BaseComponent[]`?

Reply:
We have only one TextComponent that is one BaseComponent.

Reply:
Perhaps it'd be nice to add a new method to accept `BaseComponent[]` that simply adds them as extras? This is because `ComponentBuilder#create` returns `BaseComponent[]` (as do many other methods to construct them).

Reply:
Yea, will do that after uni paper stuff.

Reply:
funny, I asked the same thing

Reply:
Yeah, but Proxy provided an example where it is reasonable.